packages/hash/task-executor/README.md

```diff
@@ -4,4 +4,12 @@ The HASH Task Executor is a temporary solution to allow the workspace to trigger
 
 Designed to be built into a self-contained docker image ([see the relevant Dockerfile](/packages/hash/docker/task-executor/Dockerfile)), the Executor is a lightweight Node HTTP server with a set of hardcoded routes to run the bundled tasks.
 
-Tasks are executed on child processes: \* Python scripts are executed within a virtual environment managed by the [./setup.sh](./setup.sh) script, with dependencies defined in [./requirements.txt](./requirements.txt)
+Tasks are executed on child processes: \* Python scripts are executed within a virtual environment managed by the [./setup.sh](./setup.sh) script, with dependencies defined in [./requirements.lock](./requirements.lock)
+
+## Updating Dependencies
+
+- Create a clean virtual environment with `python3 -m venv venv`
+- Activate it with `./venv/bin/activate`
+- Update [./requirements.txt](./requirements.txt) with the necessary top-level dependency
+- Run `python -m pip install -r ./requirements.txt`
```

Comment:
Uh I don't think this should be blocking. I believe it's correct as is.

Python and pip are silly in that they don't let you keep track of top-level dependencies. So in npm land you put in your dependencies in your package.json and then your lock file is the fully qualified tree including all of the sub-dependencies.

If you run pip freeze it will output all installed packages, and you have no way of knowing which top-level ones there are.

Because of this, my Python setup is a little verbose but lets you keep track of your top-level dependencies in a requirements.txt file, **for updating** you then install those (which installs all of the sub-dependencies), and you freeze that into a requirements.lock file.
**For normal use**, you install from the lock file and have a consistent set of dependencies.

Reply:
Ah sorry I mistakenly thought you had renamed `requirements.txt` to `requirements.lock`, thanks for explaining why they both need to exist.

Reply:
It was more self-explanatory when there were dependencies. But we moved from Singer taps to AirByte connectors (I was doing dev on the same branch and rebased) and I'd already put in the code and explanation of how to use the Python stuff, I figured leave it there as it's useful, but now a little less clear just empty requirements files.